pyccel/codegen/printing/pycode.py

```diff
@@ -697,7 +699,23 @@ def _print_IfTernaryOperator(self, expr):
         return '{true} if {cond} else {false}'.format(cond = cond, true =value_true, false = value_false)
 
     def _print_Literal(self, expr):
-        return repr(expr.python_value)
+        dtype = expr.dtype
+        precision = expr.precision
+
+        if not isinstance(expr, (LiteralInteger, LiteralFloat, LiteralComplex)) or \
+                precision == default_precision[self._print(dtype)]:
+            return repr(expr.python_value)
+        else:
+            cast_func = DtypePrecisionToCastFunction[dtype.name][precision]
+            type_name = cast_func.__name__.lower()
+            is_numpy  = type_name.startswith('numpy')
+            cast_name = cast_func.name
+            name = self._aliases.get(cast_func, cast_name)
+            if is_numpy and name == cast_name:
+                self.insert_new_import(
+                        source = 'numpy',
+                        target = cast_name)
+            return '{}({})'.format(name, repr(expr.python_value))
```

Comment:
If you do the following in python:
```
a = np.int32(3)
```
Then this is translated to the following pyccel nodes:
```
Assign(Variable(a), LiteralInteger(3, precision=4))
```
As the precision is stored in the literal object rather than saving the cast, the cast needs to be retrieved for printing in python. We can't create an actual cast node as the conversion to a literal is done in the `NumpyInt32.__new__` function.

This is therefore a bug fix for the python printer to ensure that literals are created with the correct precision

Reply:
Thanks for the explanation!